src/app/courses/pages/new-course-page/new-course-page.component.ts

```diff
@@ -10,19 +13,36 @@ import { Router } from '@angular/router';
 export class NewCoursPageComponent implements OnInit {
 
   formNewCourse!: FormGroup;
+  course: Course;
+
+  constructor(
+    private router: Router,
+    private route: ActivatedRoute,
+    private coursesService: CoursesService,
+     ) {}
 
-  constructor(private router: Router) { }
+  title: string = 'New course';
 
   ngOnInit(): void {
     this.buildForm();
+    this.route.params.subscribe(
```

Comment:
It's good practice to unsubscribe from the subscription in the ngOnDestroy().
Example: https://angular-training-guide.rangle.io/routing/routeparams
In the example above the type of property 'sub' should be defined(`private sub: Subscription;`)

StackOverflow discussion about when we should unsubscribe from observables:
https://stackoverflow.com/questions/38008334/angular-rxjs-when-should-i-unsubscribe-from-subscription